[ADD] accounting: add document for l10n_in_edi (#1940)
odoo/documentation: content/applications/finance/accounting/fiscal_localizations/localizations/india.rst

For this, go to *Apps* and search for ``l10n_in``. Then click *Install* in the module **Indian - E-invoicing**. This module has a dependency with **Indian - Accounting**. In case this last
one is not installed, Odoo installs it automatically with E-invoicing.

.. image:: media/india_edi_01.png
   :align: center
   :alt: Installation of the Indian localization module in Odoo Apps

.. note::
   When you install a database from scratch selecting India as country, Odoo will automatically install the base module: Indian - Accounting.

Setup E-invoice Service
------------------------

Once the modules are installed, to be able to start the service,
it’s necessary to set the username and password for the company.

Go to :menuselection:`Accounting --> Configuration --> Settings` and
look for the *Indian Electronic Invoicing* section.

.. image:: media/india_edi_02.png
   :align: center

.. note::
   For the creation of an API username and password please refer to this document:
   https://service.odoo.co.in/einvoice_create_api_user

Journal Configuration
----------------------

To configure you journals go to :menuselection:`Accounting --> Configuration --> Journals`:

> Open your sales journal. Under the Advanced Settings tab, in the *Electronic invoicing* field, check E-Invoice (IN) and Save. By default, it is *Checked* for sales journals.

Reviewer: could you use `:guilabel:` for elements that are on-screen and `:menuselection:` for tabs?

```suggestion
Open your sales journal. Under the :menuselection:`Advanced Settings` tab, in the
:guilabel:`Electronic Invoicing` field, check :guilabel:`E-Invoice (IN)` and Save.```

#. Enter the OTP code and click on :guilabel:`Verify OTP`.

   .. image:: india/trigger-otp.png
      :align: center
      :alt: Trigger an OTP to your registerd phone number

#. Select :guilabel:`Through GSP` in the first field, select :guilabel:`Tera Software Limited`
   as your GSP, and type in a :guilabel:`Username` and :guilabel:`Password` for your API.

   .. image:: india/submit-api-registration-details.png
      :align: center
      :alt: Submit API specific Username and Password

#. Click on :guilabel:`Submit`.

.. _india/e-invoicing-configuration:

Configuration on Odoo
---------------------

To set up the e-invoice service, go to :menuselection:`Accounting --> Configuration --> Settings -->
Indian Electronic Invoicing`, and enter the :guilabel:`Username` and :guilabel:`Password`.

.. image:: india/e-invoice-setup.png
   :align: center
   :alt: Setup e-invoice service

.. _india/e-invoicing-journals:

Journals
~~~~~~~~

To configure your journals, go to :menuselection:`Accounting --> Configuration --> Journals`. Open
your *sales* journal, and in the :guilabel:`Advanced Settings` tab, under :guilabel:`Electronic Data

> Interchange`, check :guilabel:`E-Invoice (IN)` and :guilabel:`Save`.

Reviewer 2: Should not this be automatically the case when you install Indian accounting @nia-odoo @jva-odoo

Reviewer: I think so indeed, but only for their main "tax invoice" journal, no? I'll just rephrase it so they can "double-check" and we can always improve this doc later :)

Once you request for canceling the e-invoice, Odoo will automatically submit the JSON Signed file to the government portal. But, if you want to process the invoice immediately, you can process it by clicking on :guilabel:`Process Now`.

Others
------

.. _india/verify-e-invoice:

Verify E-invoice
~~~~~~~~~~~~~~~~

After submitting E-invoice, you can also verify the signed invoice from the government portal.

#. Download the json file from the attachment.

#. Open E-invoice portal: https://einvoice1.gst.gov.in/
   go to :menuselection:`Search --> Verify Signed Invoice`

#. Browse json file and submit

   .. image:: india/verify-invoice.png
      :align: center
      :alt: browse json file for verify invoice

#. you can check verified signed E-invoice here

   .. image:: india/signed-invoice.png
      :align: center
      :alt: verified e-invoice

.. _india/e-invoice-api_registration:

E-Invoice API Registration
~~~~~~~~~~~~~~~~~~~~~~~~~~

#. Login to NIC E-Invoice Web Portal on - https://einvoice1.gst.gov.in/

   .. note::
      If you have already registered on NIC Eway Bill Production portal then you may use the same Login credentials here on - https://einvoice1.gst.gov.in/

> #. Click on Login and enter your Username and Password

Reviewer 3: ```suggestion
#. Click on :guilabel:`Login` and enter your Username and Password
```
:guilabel: must be used everytime it's a graphical element (elements that can be seen by the user).

   .. image:: india/e-invoice-system-login.png
      :align: center
      :alt: e-invoice system login

> #. You will be able to see similar screen after login. Now, click on :guilabel:`API Registration`

Reviewer 3: ```suggestion
#. After logging in, you are presented with this screen. Now, click on :guilabel:`API Registration`
```
We try to avoid the use of the future tense. Suggestion would be better,

   .. image:: india/api-registration.png
      :align: center
      :alt: Click on API registration

#. Click on :guilabel:`User Credentials`. After that, click on :guilabel:`Create API User`.

   .. image:: india/e-invoice-create-api-user.png
      :align: center
      :alt: Click on User Credentials and Create API User

> #. It will trigger an OTP to your registered mobile Number.

Reviewer 3: Avoid the future tense

   .. image:: india/trigger-otp.png
      :align: center
      :alt: Trigger an OTP to your registerd phone number

> #. Please enter OTP and click on Verify OTP.

Reviewer 3: guilabel if necessary, and no need to use "please" :)

#. Click on :guilabel:`Through GSP` and select :guilabel:`Terasoft Software Limited` as your GSP.

   .. image:: india/select-gsp.png
      :align: center
      :alt: Select your GSP

> #. Create API specific Username and Password & Click on Submit.

Reviewer 3: guilabels

   .. image:: india/submit-api-registration-details.png
      :align: center
      :alt: Submit API specific Username and Password

> #. Please confirm acknowledgement from E-Invoice System.

Reviewer 3: No need for "please"

.. image:: india/e-invoice-setup.png
   :align: center
   :alt: Setup e-invoice service

.. _india/e-invoicing-journal-configuration:

Journal configuration
~~~~~~~~~~~~~~~~~~~~~

To configure your journals, go to :menuselection:`Accounting --> Configuration --> Journals`.
Open your 'sales' type journal, and in the :guilabel:`Advanced Settings` tab, under
:guilabel:`Electronic Data Interchange`, check :guilabel:`E-Invoice (IN)` and
:guilabel:`Save`.

.. image:: india/journal-configuration.png
   :align: center
   :alt: Journal configuration

.. _india/e-invoicing-workflow:

Workflow
--------

To start invoicing from Odoo, an invoice must be created using the standard invoicing flow,
that is, either from a sales order or the invoice menu in the Accounting application

.. seealso::
   - `Odoo Tutorials: Sales Tutorials <https://www.odoo.com/slides/sales-17>`_
   - `Odoo Tutorials: Accounting <https://www.odoo.com/slides/accounting-19>`_

Invoice validation
~~~~~~~~~~~~~~~~~~

Once the invoice is validated, the following message is displayed at the top:

.. image:: india/e-invoice-process.png
   :align: center
   :alt: Click on Process Now to submit the invoice immediately

> 

Reviewer 3: There's an extra line
